bitcoin/src/crypto/key.rs

```diff
@@ -863,61 +927,73 @@ impl From<TweakedKeypair> for TweakedPublicKey {
     #[inline]
     fn from(pair: TweakedKeypair) -> Self { TweakedPublicKey::from_keypair(pair) }
 }
-/// A key-related error.
+
+/// Error generated from WIF key format.
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
-pub enum Error {
+pub enum FromWifError {
     /// A base58 error.
     Base58(base58::Error),
     /// A secp256k1 error.
     Secp256k1(secp256k1::Error),
+}
+
+/// Error returned while generating key from slice.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum FromSliceError {
     /// Invalid key prefix error.
     InvalidKeyPrefix(u8),
-    /// Hex decoding error.
-    Hex(hex::HexToArrayError),
-    /// `PublicKey` hex should be 66 or 130 digits long.
-    InvalidHexLength(usize),
+    /// A Secp256k1 error,
+    Secp256k1(secp256k1::Error),
+    /// Invalid Length of the slice
+    InvalidLength(usize),
```

Comment:
In 4717cb9a384afc21817a1fba35573311e4ce2b0b:

I'm tempted to drop this variant and just let libsecp handle invalid lenghts. @Kixunil what do you think?

Reply:
> if we just had `secp256k1` methods accept arrays

Out of scope for this PR ofc, but we can certainly do that. The reason we don't have that now is that arrays are kinda unergonomic and I was lazy.

Reply:
Sure. I think we can use `secp256k1` variant and change it later.

Reply:
Libspec does not offer the `InvalidLength` error by itself. These are the errors that are associated with it :

```rust
pub enum Error {
    /// Signature failed verification.
    IncorrectSignature,
    /// Bad sized message ("messages" are actually fixed-sized digests [`constants::MESSAGE_SIZE`]).
    InvalidMessage,
    /// Bad public key.
    InvalidPublicKey,
    /// Bad signature.
    InvalidSignature,
    /// Bad secret key.
    InvalidSecretKey,
    /// Bad shared secret.
    InvalidSharedSecret,
    /// Bad recovery id.
    InvalidRecoveryId,
    /// Tried to add/multiply by an invalid tweak.
    InvalidTweak,
    /// Didn't pass enough memory to context creation with preallocated memory.
    NotEnoughMemory,
    /// Bad set of public keys.
    InvalidPublicKeySum,
    /// The only valid parity values are 0 or 1.
    InvalidParityValue(key::InvalidParityValue),
    /// Bad EllSwift value
    InvalidEllSwift,
}
```
I think the most relevant would be to use the variant `InvalidPublicKey` and something like :
``Err(FromSliceError::Secp256k1(secp256k1::Error::InvalidPublicKey))``

Reply:
@Harshit933 if you try to parse a key of the wrong length you'll just get an `InvalidPublicKey` error.

My point was that we don't actually need to check the length because the libsecp parser does it for us. The reason that we *do* check the length is so that we can determine compressedness (and exclude hybrid keys). But in doing so, we provide a redundant error path.

Reply:
Oh I thought there's a dedicated variant. In that case I would prefer having one here since we can provide more context which helps debugging. Or add it to secp, but that would take longer.

Reply:
@apoelstra I got your point, For now I'll keep it as it is as suggested by @Kixunil .